week3/src/components/LevelButton.jsx

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable react/prop-types */
+import styled from "styled-components";
+
+function LevelButton({ setCards, setDifficulty, getMixedCardList }) {
+  const handleEasy = () => {
+    setCards(getMixedCardList(5));
+    setDifficulty("easy");
+  };
+
+  const handleNormal = () => {
+    setCards(getMixedCardList(7));
+    setDifficulty("normal");
+  };
+
+  const handleHard = () => {
+    setCards(getMixedCardList(9));
+    setDifficulty("hard");
+  };
+
+  return (
+    <>
+      <ButtonWrapper>
+        <EasyBtn onClick={handleEasy}>Easy</EasyBtn>
+        <NormalBtn onClick={handleNormal}>Normal</NormalBtn>
+        <HardBtn onClick={handleHard}>Hard</HardBtn>
+      </ButtonWrapper>
+    </>
+  );
+}
+
+// button
+const ButtonWrapper = styled.div`
+  display: flex;
+  gap: 2vw;
+  justify-content: center;
+  align-items: center;
+  margin: 15vh 0 8vh 0;
+`;
+const EasyBtn = styled.button`
+  width: 10vw;
+  height: 4vh;
+  background-color: ${({ theme }) => theme.colors.lightYellow};
+  border: none;
+  border-radius: 30px;
+  &:hover {
+    background-color: ${({ theme }) => theme.colors.skyBlue};
+  }
+`;
+const NormalBtn = styled.button`
+  width: 10vw;
+  height: 4vh;
+  background-color: ${({ theme }) => theme.colors.lightYellow};
+  border: none;
+  border-radius: 30px;
+  &:hover {
+    background-color: ${({ theme }) => theme.colors.skyBlue};
+  }
+`;
+const HardBtn = styled.button`
+  width: 10vw;
+  height: 4vh;
+  background-color: ${({ theme }) => theme.colors.lightYellow};
+  border: none;
+  border-radius: 30px;
+  &:hover {
+    background-color: ${({ theme }) => theme.colors.skyBlue};
+  }
+`;
```

Comment:
p5: 난이도 별로 버튼을 만드셨군요~ 공통되는 스타일이 많은데, 하나의 버튼 스타일을 만들어놓고 상속하는 방법도 소개해드릴게요~ 

```suggestion
const Button = styled.button`
  // ...
`;

const EasyButton = styled(Button)`
  // ...
`;

const NormalButton = styled(Button)`
  // ...
`;

const HardButton = styled(Button)`
  // ...
`;

```

Reply:
공통된 스타일 속성을 만들고 그걸 상속받게도 할 수 있네요!!
유용한 정보 알아갑니다 !! 👍